core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/OutputManager.scala

```diff
@@ -107,15 +108,12 @@ class OutputManager(
     )
   }
 
-  def emitEpochMarker(epochMarker: EpochMarker): Unit = {
-    // find the network output ports within the scope of the marker
-    val outputsWithinScope =
-      networkOutputBuffers.filter(out => epochMarker.scope.links.map(_.id).contains(out._1._1))
-    // flush all network buffers of this operator, emit epoch marker to network
-    outputsWithinScope.foreach(kv => {
-      kv._2.flush()
-      kv._2.addEpochMarker(epochMarker)
-    })
+  def flush(onlyFor: Option[Set[PhysicalLinkIdentity]] = None): Unit = {
```

Comment:
Also, I would suggest use 
```suggestion
  def flush(onlyFor: Set[PhysicalLinkIdentity] = Set()): Unit = {
      if (onlyFor.empty){
      send all
      } else {
       send only for
      }
  }
```

Reply:
This change will affect the logic of this function. I want to have a case for `onlyFor` to be an empty set and we don't flush for any of the channels.